### PR TITLE
LRDOCS-7169 7.2 architecture intro

### PR DESCRIPTION
**Pull request by jhinkey**
Here's the architecture intro. I'll add links to it as I add sub articles to.

https://issues.liferay.com/browse/LRDOCS-7169

### CODE REVIEW

File: developer/customization/articles/03-architecture/01-architecture-intro.markdown [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+header-id: architecture
+---
+
+# Architecture
+
+[TOC levels=1-4]
+
+@product@'s architecture helps you build sites that offer powerful functionality
+and terrific styling, by providing these benefits and more:
+
```
Comment (sez11a) [inline developer/customization/articles/03-architecture/01-architecture-intro.markdown]: Architecture is architecture; it doesn't help you build things. You would want to start with how well-designed Liferay's architecture is, what the parts are, and how they fit together seamlessly to provide a whole that is greater than the sum of its parts.

File: developer/customization/articles/03-architecture/01-architecture-intro.markdown [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+- Leverages development standards
+
+- Facilitates swapping implementations
+
+- Starts up fast and performs well
+
+- Its runtime is easy to inspect and configure
+
+The architecture comprises these components: 
+
```
Comment (sez11a) [inline developer/customization/articles/03-architecture/01-architecture-intro.markdown]: Yes; I'd start with this information.

File: developer/customization/articles/03-architecture/01-architecture-intro.markdown [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+**Services:** Functionality exposed via Java APIs and web APIs. 
+
+**Core:** Web application that bootstraps @product@ and provides a runtime 
+framework for deploying components and customizations. 
+
+@product@ and Liferay apps provide the UI and services that users and external
+components interact with. The UIs and services are the parts you can customize
+in @product@ and add to it. Liferay services do the "heavy lifting". 
+
```
Comment (sez11a) [inline developer/customization/articles/03-architecture/01-architecture-intro.markdown]: TMI. In place of the paragraph above should be a nice diagram fitting the parts together, and then you can illustrate the diagram. Be sure to include project logos in the diagram (OSGi and Spring in the core; services we use---such as Apache CXF or OpenSAML---for Services, and PortletMVC4Spring, JSF, React, Vue, and Angular in the UIs.

Comment (sez11a) [reply developer/customization/articles/03-architecture/01-architecture-intro.markdown]: See our previous efforts at this: 
https://portal.liferay.dev/docs/7-1/tutorials/-/knowledge_base/t/fundamentals#technologies

https://portal.liferay.dev/docs/7-1/tutorials/-/knowledge_base/t/fundamentals#architecture

**Comment by jhinkey** [comment]
I'll send a new pull after addressing these requests.

**Pull request closed without merging**